### PR TITLE
Implemented the search feature

### LINKED ISSUE #333: Implement search options
When searching nothing happens.

### PR DESCRIPTION
**Pull request by duke7553**
For code review purposes. We implement search which allows for recursive queries inside any normal directory. We also aim to support hidden items with this PR.

Limitations: Only indexed locations are queried for performance reasons. We'll add a toggle to change this in a future PR

Fixes #333

### CODE REVIEW

**Comment by yaichenbaum** [comment]
- It looks like the search box cuts off other items in the navigation bar when you make the windows size smaller.
- We should hide and clear the text in the search box if it loses focus.

File: Files/BaseLayout.cs [hunk 1]
```diff
@@ -242,46 +242,61 @@ private void NotifyPropertyChanged([CallerMemberName] string propertyName = "")
             PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
         }
 
+        bool IsSearchResultPage = false;
```
Comment (d2dyno1) [inline Files/BaseLayout.cs:245]: Move this variable to the top?

**Comment by d2dyno1** [comment]
Hmmm.. It seems like it does not display any search results for me no matter what I search for

**Comment by Sank30** [comment]
Search logo shown as a greyed out for my every location.

File: Files/BaseLayout.cs (hunk 1 above)
Comment (duke7553) [reply Files/BaseLayout.cs]: @d2dyno1 Thanks! Good catch.

**Review by yaichenbaum — APPROVED** [review]

**Review by yaichenbaum — APPROVED** [review]

**Comment by yaichenbaum** [comment]
@duke7553 Great work!

**Pull request merged**